src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrderCheck.java

```diff
@@ -0,0 +1,438 @@
+ * </p>
+ * <pre>
+ * class example2 {
+ *   &#47;**
+ *    * No violation for missing "str" parameter.
+ *    *
+ *    * &#64;version Some javadoc.
+ *    * &#64;param &lt;T&gt; description
+ *    * &#64;param E desc          // violation, wrong order for "E" tag
+ *    * &#64;param &lt;S&gt; desc
+ *    * &#64;param array desc
+ *    * &#64;see Some javadoc
+ *    * &#64;param e desc          // violation, wrong order for "e" tag
+ *    * &#64;return Some number.
+ *    *&#47;
+ *   public static &lt;T, E, S&gt; int sampleMethod(T[] array, String str, E e) {
+ *     return 2021;
+ *   }
+ *
+ *   &#47;**
+ *    * No violation for missing "output" parameter.
+ *    *
+ *    * &#64;param input desc      // ok
+ *    *&#47;
+ *   public void sampleMethod2(String input, String output) {
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * This check will not report any violations for extra parameter in javadoc.
+ * </p>
+ * <pre>
+ * class example3 {
+ *   &#47;**
+ *    * No violation for extra "extra" parameter.
+ *    *
+ *    * &#64;version Some javadoc.
+ *    * &#64;param &lt;T&gt; description
+ *    * &#64;param E desc          // violation, wrong order for "E" tag
+ *    * &#64;param &lt;S&gt; desc
+ *    * &#64;param array desc
+ *    * &#64;see Some javadoc
+ *    * &#64;param e desc          // violation, wrong order for "e" tag
+ *    * &#64;param str desc        // violation, wrong order for "str" tag
+ *    * &#64;param extra desc      // ok, Extra parameter
+ *    * &#64;return Some number
+ *    *&#47;
+ *   public static &lt;T, E, S&gt; int sampleMethod(T[] array, String str, E e ) {
+ *     return 2021;
+ *   }
+ *
+ *   &#47;**
+ *    * No violation for extra "t" parameter.
+ *    *
+ *    * &#64;param t desc          // ok, Extra parameter
+ *    *&#47;
+ *   public void sampleMethod2() {}
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+  * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.tag.order}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 8.43
+ */
+@StatelessCheck
+public class JavadocParamOrderCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_JAVADOC_TAG_ORDER = "javadoc.tag.order";
+
+    /**
+     * Specify the list.
+     */
+    private final List<Integer> targetParent = Arrays.asList(
+        TokenTypes.INTERFACE_DEF,
+        TokenTypes.RECORD_DEF,
+        TokenTypes.MODIFIERS,
+        TokenTypes.TYPE,
+        TokenTypes.TYPE_PARAMETERS,
+        TokenTypes.CTOR_DEF
+    );
+
+    /**
+     * Specify the list.
+     */
+    private final List<String> stringList = new ArrayList<>();
```

Comment:
why it can be local variable? It used just a container for values between two methods